test/Microsoft.Sbom.Targets.Tests/GenerateSbomTaskTests.cs

```diff
@@ -5,37 +5,257 @@
+    private static readonly string TemporaryDirectory = Path.Combine(CurrentDirectory, "_temp");
+    private const string PackageSuplier = "Test-Microsoft";
+    private const string PackageName = "CoseSignTool";
+    private const string PackageVersion = "0.0.1";
+    private const string NamespaceBaseUri = "https://base0.uri";
+
     private Mock<IBuildEngine> buildEngine;
     private List<BuildErrorEventArgs> errors;
-    private static readonly string CurrentDirectory = Path.GetDirectoryName(System.Reflection.Assembly.GetExecutingAssembly().Location);
-    private static readonly string ManifestDirectory = Path.Combine(CurrentDirectory, "_manifest");
+    private ISBOMValidator sbomValidator;
+    private string manifestPath;
+    private SbomSpecification sbomSpecification;
+    private GeneratedSbomValidator generatedSbomValidator;
+
+    private string SbomSpecificationDirectoryName => $"{this.sbomSpecification.Name}_{this.sbomSpecification.Version}";
+
+    public GenerateSbomTaskTests()
+    {
+        this.sbomSpecification = Constants.SPDX22Specification;
+    }
 
     [TestInitialize]
     public void Startup()
     {
+        var host = Host.CreateDefaultBuilder()
+            .ConfigureServices((host, services) =>
+                services
+                .AddSbomTool())
+            .Build();
+        this.sbomValidator = host.Services.GetRequiredService<ISBOMValidator>();
+
         // Setup the build engine
-        buildEngine = new Mock<IBuildEngine>();
-        errors = new List<BuildErrorEventArgs>();
-        buildEngine.Setup(x => x.LogErrorEvent(It.IsAny<BuildErrorEventArgs>())).Callback<BuildErrorEventArgs>(e => errors.Add(e));
+        this.buildEngine = new Mock<IBuildEngine>();
+        this.errors = new List<BuildErrorEventArgs>();
+        this.buildEngine.Setup(x => x.LogErrorEvent(It.IsAny<BuildErrorEventArgs>())).Callback<BuildErrorEventArgs>(e => errors.Add(e));
+
+        // Clean up the manifest directory
+        if (Directory.Exists(DefaultManifestDirectory))
+        {
+            Directory.Delete(DefaultManifestDirectory, true);
+        }
 
         // Clean up the manifest directory
-        if (Directory.Exists(ManifestDirectory))
+        if (Directory.Exists(TemporaryDirectory))
+        {
+            Directory.Delete(TemporaryDirectory, true);
+        }
+
+        this.manifestPath = Path.Combine(DefaultManifestDirectory, this.SbomSpecificationDirectoryName, "manifest.spdx.json");
+        this.generatedSbomValidator = new(this.sbomSpecification);
+    }
+
+    [TestCleanup]
+    public void Cleanup()
+    {
+#pragma warning disable VSTHRD002 // Avoid problematic synchronous waits
+        if (File.Exists(this.manifestPath))
         {
-            Directory.Delete(ManifestDirectory, true);
+            Assert.IsTrue(this.sbomValidator.ValidateSbomAsync(
+                CurrentDirectory,
+                TemporaryDirectory,
```

Comment:
Should this be DefaultManifestDirectory instead as most of the tests use the default _manifest directory? I noticed  Sbom_Is_Successfully_Generated_In_Specified_Location() is the only test that utilizes the TemporaryDirectory as the target manifest location.